_im-plugin/append-only-index.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: default
+title: Append-only index
+nav_order: 14
+---
+
+# Append-only index
+
+An append-only index is an immutable index that only allows document ingestion (appending) while blocking all updates or deletions after initial document creation. When you enable the append-only setting for an index, OpenSearch prevents any modifications to existing documents. You can only add new documents to the index.
+
+When you configure an index as append-only, the following operations return an error:
+
+- Document update call (Update API)
+- Document delete call (Delete API)
+- Update by query call
+- Delete by query call
+- Bulk API calls made with the update, delete, or upsert actions
+- Bulk API calls containing an index action with a custom document ID
+
+## Benefits
+
+Append-only indexes offer several advantages:
+
+- Optimized performance by eliminating costly update and delete operations
+- Optimized storage and segment merges by eliminating soft deletes and version tracking
+- Support for future optimizations like auto-rollovers and efficient warm tiering
+
+Append-only indexes are ideal for immutable workloads, such as those containing log, metric, observability, or security event data, where data is not modified once ingested.
+
+## Creating an append-only index
+
+The following request creates a new index named `my-append-only-index` with all updates disabled:
+
+```json
+PUT /my-append-only-index
+{
+  "settings": {
+    "index.append_only.enabled": true
+  }
+}
+```
+{% include copy-curl.html %}
+
+After an index is set to append-only, it cannot be changed to another index type.
+{: .warning}
+
+
+If you want to append data from an existing index to the new append-only index, use the Reindex API. The following example reindexes documents from a source index (`logs`) into the new append-only index:
+
+```json
+POST /_reindex
+{
+  "source": {
+    "index": "my-source-index"
+  },
+  "dest": {
+    "index": "my-append-only-index"
```

Comment:
Earlier change was correct. This looks wrong.

@Naarcha-AWS @RS146BIJAY Looks like changes are being overridden. Could you please see on how to fix them?

Reply:
@mgodwan: This is the reindexing call and is the same change I made before the override. @RS146BIJAY, did you modify this? If so, I can't see your modification.

Reply:
@Naarcha-AWS I think I had earlier added the right changes. This seems to have been modified by a recent commit.

Reply:
@Naarcha-AWS  Yes I modified it as per @mgodwan  comment. Let me push the changes again.

Reply:
@RS146BIJAY: Don't. I already fixed it.